src/cim/components/ClusterDeployment/ClusterDeploymentDetails.tsx

```diff
@@ -0,0 +1,217 @@
+  shouldShowClusterInstallationError,
+  shouldShowClusterInstallationProgress,
+} from './helpers';
+import ClusterProgress from '../../../common/components/clusterDetail/ClusterProgress';
+import ClusterInstallationError from './ClusterInstallationError';
+import { EventsModalButton } from '../../../common/components/ui/eventsModal';
+import KubeconfigDownload from '../../../common/components/clusterDetail/KubeconfigDownload';
+import { EventListFetchProps } from '../../../common/types/events';
+import AgentTable from '../Agent/AgentTable';
+
+const agentTableColumns = [
+  { title: 'Hostname', transforms: [sortable], cellFormatters: [expandable] },
+  { title: 'Role', transforms: [sortable] },
+  { title: 'Status', transforms: [sortable] },
+  { title: 'Discovered At', transforms: [sortable] },
+  { title: 'CPU Cores', transforms: [sortable] }, // cores per machine (sockets x cores)
+  { title: 'Memory', transforms: [sortable] },
+  { title: 'Disk', transforms: [sortable] },
+  { title: '' },
+];
+
+type ClusterDeploymentDetailsProps = {
+  clusterDeployment: ClusterDeploymentK8sResource;
+  agentClusterInstall: AgentClusterInstallK8sResource;
+  agents: AgentK8sResource[];
+  downloadKubeconfig: () => Promise<void>;
+  fetchEvents: EventListFetchProps['onFetchEvents'];
+  fetchCredentials: (setCredentials: unknown) => Promise<void>;
+  agentTableClassName?: string;
+};
+
+const ClusterDeploymentDetails: React.FC<ClusterDeploymentDetailsProps> = ({
+  clusterDeployment,
+  agentClusterInstall,
+  agents,
+  downloadKubeconfig,
+  fetchEvents,
+  fetchCredentials,
+  agentTableClassName,
+}) => {
+  const [progressCardExpanded, setProgressCardExpanded] = React.useState(true);
+  const [inventoryCardExpanded, setInventoryCardExpanded] = React.useState(true);
+  const [detailsCardExpanded, setDetailsCardExpanded] = React.useState(true);
+
+  const cluster = getAICluster({ clusterDeployment, agentClusterInstall, agents });
+  const [clusterStatus, clusterStatusInfo] = getClusterStatus(agentClusterInstall);
+  return (
+    <Stack hasGutter>
+      {shouldShowClusterInstallationProgress(agentClusterInstall) && (
+        <StackItem>
+          <Card id="cluster-installation-progress-card" isExpanded={progressCardExpanded}>
+            <CardHeader
+              onExpand={() => setProgressCardExpanded(!progressCardExpanded)}
+              toggleButtonProps={{
+                id: 'progress-card-toggle-button',
+                'aria-label': 'Cluster installation process',
+                'aria-labelledby': 'titleId progress-card-toggle-button',
+                'aria-expanded': progressCardExpanded,
+              }}
+            >
+              <CardTitle id="titleId">Cluster installation process</CardTitle>
+            </CardHeader>
+            <CardExpandableContent>
+              <CardBody>
+                <Stack hasGutter>
+                  <StackItem>
+                    <ClusterProgress
+                      cluster={cluster}
```

Comment:
As discussed, we'll add to the ClusterProgress a prop called `totalProgress` that twill allow the parent component to pass the calculation if necessary